pgtrigger/compiler.py

```diff
@@ -1,9 +1,16 @@
+from __future__ import annotations
+
 import collections
 import hashlib
+from typing import Any, Dict, List, NoReturn, Optional, Tuple, Union, NewType
 
+import pgtrigger
 from pgtrigger import utils
 
-_unset = object()
+
+Unset = NewType("Unset", object)
```

Comment:
NewType helps prevents bugs by accepting `object`. Accepting `object` as a type-hint significantly reduces the type-safe of a argument, so this enforces that we explicitly are setting it our unset object. Before having this I had actually incorrectly typed some things.